### ISSUE TITLE
Question / Feature request - How to share bundled code with multiple entry points

### ISSUE THREAD

**Issue by choksheak** [evanw/esbuild#929]
Hi, I am currently using esbuild on a project with multiple entry point html files. esbuild is a huge time-saver for me!! As one could imagine, these entry points basically start off by rendering a different webpage, but 99% of the code is identical. Therefore I am wondering whether it is possible for esbuild to automatically generate all the shared code as one bundled JS, and then have the entry points import the bootup file (render entry view) and the bundled JS? I know this can be done manually but since this sounds like a somewhat common use case (for non-SPAs) so I thought having tooling support for this would be too awesome.

E.g. current:
```
view1.html (300 bytes) -> import bundled module view1.js
view1.js (1MB) -> all bundled JS
view2.html (300 bytes) -> import bundled module view2.js
view2.js (1MB) -> all bundled JS
Total: 2MB+
```

E.g. shared bundle:
```
view1.html (300 bytes) -> import bundled module bundle.js + bootup view1.js
view1.js (100 bytes) -> render startup view
bundled.js (1MB) -> all bundled JS
view2.html (300 bytes) -> import bundled module bundle.js + bootup view2.js
view2.js (100 bytes) -> render startup view
Total: 1MB+
```

And you can see, each additional entry point essentially duplicates the entire bundled JS if not shared.

Look forward to hearing back from you. Thank you! 😄

**Comment by evanw** [comment]
Is this solved by enabling [code splitting](https://esbuild.github.io/api/#splitting)? That is the intent of the code splitting feature, although it is still a work in progress.

**Comment by choksheak** [comment]
Just tried and got `error: Splitting currently only works with the "esm" format`. Didn't realize this is already work-in-progress, but now I know! I think we can try our manual workaround before this becomes ready for use. Thank you!

**Comment by choksheak** [comment]
Hi, just wanted to follow-up and share what we ended up doing.

1. Create a new file bundle.tsx.
2. In `bundle.tsx`, create a switch statement based on the window.location.pathname. The pathname will correctly determine which page to render based on which html page is put in the browser's address bar.
3. All the html entrypoint files will now be empty stubs that include the compiled bundle.js file using the `<script>` tag.

This approach is simple and yet very effective! We ended up halving our entire bundle size by doing this.

To illustrate a bit more details using code (everything here is simplified for illustration purposes only):

**bundle.tsx:**
```
switch (window.location.pathname) {
  case "/index.html":
    return render(<Home />);
  case "/page1.html":
    return render(<Page1 />);
  default:
    return <PageNotFound />;
}
```

**index.html:**
```
<!DOCTYPE html>
<title>Best Website Ever</title>
<div id="root"></div>
<script type="module" src="bundle.js"></script>
```

Hope this is helpful. :)
